<?xml version="1.0"?>

<launch>

    <arg name="veh" default="wamv" />
    <arg name="cmd" default="cmd_vel" />
    <arg name="joy" default="/joy" />
    <arg name="mission" default="path_following" />
    <arg name="goal_dis" default="12.5" />
    <arg name="target_angle" default="0.2" />
    <arg name="frame" default="map" />
    <arg name="linear_scale" default="0.3" />
    <arg name="angular_scale" default="0.45" />

    <!-- <include file="$(find subt_rl)/launch/pcToLaser_240_sim_wamv.launch">
        <arg name="veh" value="$(arg veh)" />
    </include> -->

    <!-- map truth_map_posestamped -->
    <group ns="$(arg veh)">

        <node pkg="wamv_rl" type="goal_nav_tf_vel_wamv.py" name="goal_nav_wamv_vel_rl" output="screen">
            <param name="frame" value="$(arg frame)" />
            <param name="goal_dis" value="$(arg goal_dis)" />
            <param name="mission" value="$(arg mission)" />
            <param name="linear_scale" value="$(arg linear_scale)" />
            <param name="angular_scale" value="$(arg angular_scale)" />
            <remap from="goal_in" to="/move_base_simple/goal" />
            <remap from="laser_in" to="RL/scan" />
            <remap from="odom_in" to="localization_gps_imu/pose" />
            <remap from="joy" to="$(arg joy)" />
            <remap from="cmd_out" to="$(arg cmd)" />
        </node>

        <node pkg="wamv_rl" type="scan_angle_alignment.py" name="scan_angle_alignment" output="screen">
            <param name="frame" value="$(arg frame)" />
            <param name="target_angle" value="$(arg target_angle)" />
        </node>

    </group>


</launch>
